bazel/foreign_cc/BUILD

```diff
@@ -169,7 +169,7 @@ envoy_cmake(
         # FindOpenSSL.cmake which fails at what looks like version parsing
         # (the libraries are found ok).
         "CURL_CA_PATH": "none",
-        "CMAKE_USE_OPENSSL": "off",
+        "CURL_USE_OPENSSL": "off",
```

Comment:
Based on the comment above and the fact that we've updated build toolchains to a more modern flavor of cmake, and curl and openssl have made tweaks here, this may no longer be true - we might not have to dodge this anymore. @lizan when you are back could you comment?

Reply:
@wrowe without this tests are not passing. what do you recommend as the way forward?